libs/market-list/src/lib/components/markets-container/market-list-table.tsx

```diff
@@ -100,12 +96,22 @@ export const MarketListTable = forwardRef<AgGridReact, Props>((props, ref) => {
         type="rightAligned"
         cellRenderer="PriceFlashCell"
         filter="agNumberColumnFilter"
-        valueGetter={({ data }: { data?: MarketWithData }) => {
+        valueGetter={({
+          data,
+        }: VegaValueGetterParams<
+          MarketWithData,
+          NonNullable<MarketWithData['data']>['bestBidPrice']
```

Comment:
question (blocking): Why use NonNullable here? Isn't that unsafe with AgGrid? Also the type helper should work with dot notation strings. See below, right now if you pull in `value` its type is lost

```
VegaValueGetterParams<MarketWithData, 'data.bestBidPrice'>
```

Same for other other parts of this file

Reply:
I wrongly used VegaValueGetterParams here, instead of passing key I passed field type.

`NonNullable<MarketWithData['data']>['bestBidPrice']` is taking exact type of bestBidPrice. NonNullable is needed to take 'data' property type from MarketWithData without null and undefined.
It's not possible to do MarketWithData['data']['bestBidPrice']

I see one problem with passing key and using type fest Get. If you will provide wrong key, type will be unknown. If field will not exist any more because source type changed it will be not recognised by typescript.